examples/step-40/step-40.cc

```diff
@@ -693,17 +690,27 @@ namespace Step40
 // @sect4{main()}
 
 // The final function, <code>main()</code>, again has the same structure as in
-// all other programs, in particular step-6. Like in the other programs that
-// use PETSc, we have to initialize and finalize PETSc, which is done using the
-// helper object MPI_InitFinalize.
+// all other programs, in particular step-6. Like the other programs that use
+// PETSc, we have to initialize and finalize PETSc, which is done using the
+// helper object Utilities::MPI::MPI_InitFinalize. The constructor of that
+// class initializes MPI other libraries that depend on it, such as p4est,
+// SLEPc, and Zoltan (though the last two are not used in this tutorial). The
+// order here is important: we cannot use any of these libraries until they
+// are initialized, so it does not make sense to do anything before creating
+// an instance of Utilities::MPI::MPI_InitFinalize.
 //
-// Note how we enclose the use the use of the LaplaceProblem class in a pair
-// of braces. This makes sure that all member variables of the object are
-// destroyed by the time we destroy the mpi_initialization object. Not doing
-// this will lead to strange and hard to debug errors when
-// <code>PetscFinalize</code> first deletes all PETSc vectors that are still
-// around, and the destructor of the LaplaceProblem class then tries to delete
-// them again.
+// After the solver finishes, the LaplaceProblem destructor will run followed
+// by Utilities::MPI::MPI_InitFinalize::~MPI_InitFinalize(). This order is
+// also important: Utilities::MPI::MPI_InitFinalize::~MPI_InitFinalize() calls
+// <code>PetscFinalize</code> (and finalization functions for other
+// libraries), which will delete any in-use PETSc objects. This must be done
+// after we destruct the Laplace solver to avoid double deletion
+// errors. Fortunately, due to the order of destructor call rules of C++, we
+// do not need to worry about any of this: everything happens in the correct
+// order (i.e., the reverse of the order of construction). The last function
+// called by Utilities::MPI::MPI_InitFinalize::~MPI_InitFinalize() is
+// <code>MPI_Finalize</code>: i.e., once this object is destructed the program
+// should exit since MPI will no longer be available.
```

Comment:
I couldn't figure out what the original text meant by 'pair of braces' so I just rewrote the whole thing.

Reply:
Sounds good!